### ISSUE TITLE
Permissions for regular user php8.0 joomla 4 dev beta 8

### ISSUE THREAD

**Issue by breebee** [akeeba/engage#76]
**Steps to reproduce the issue:**
_____________ 

- Create a regular "registered" group user. Attempt to comment on article

--------

**Expected result:**
_____________

- Comment gets posted

-------

**Actual result:**
____________

- Regular registered user gets redirected to a 403 by dispatcher after posting a comment. Only super users can comment

--------

**Troubleshooting already performed:**
____________
- Double checked all permissions for users/usergroups
- Disabled all other component plugins
- Reinstalled engage removing all folders and database tables
- Tested cassiopeia template with no overrides 
--------

____________
System information
--------
    Joomla! version: 4.beta.8.dev
    PHP version: 8.0
    Am I using HTTPS: No ( localhost )
    Server Operating System: Linux
    My device's Operating System: Linux
    My device's Operating System version: Ubuntu 18.04
    Browser: Firefox
    Browser version: 86.0b5 64bit

    MySQL version: 10.1.47-MariaDB-0ubuntu0.18.04.1
    Server type: NginX
    Server version: 1.14.0
    Hosting company / Local server: local

**Comment by breebee** [comment]
[
![Screenshot_2021-03-22 Errore 403](https://user-images.githubusercontent.com/13717576/112010946-10b23800-8b28-11eb-8436-e0a2da8bfd48.png)
](url)

**Comment by nikosdion** [comment]
If only I had documented this... Oh, wait, I did. https://github.com/akeeba/engage/wiki/Installation-and-Setup-Guide#tell-akeeba-engage-who-can-submit-and-manage-comments

**Comment by nikosdion** [comment]
BTW, if you want to see plain Registered users being able to comment on articles you just need to go to my blog, https://www.dionysopoulos.me which is indeed running Joomla 4 and has comments enabled for Registered users. You can register a user using the social login buttons, including GitHub, on the sidebar of the site. You can then comment on any blog post.

What you described is, therefore, the exact use case this component was written for. That's why I documented it explicitly.

Would it make sense to allow Registered users to comment by default? Sure! However, you cannot tell Joomla to set custom permissions when installing an extension. It makes sense, though. You can use any group IDs and names you want. Therefore a developer wouldn't know where to set custom permissions. And that's before you start considering other nuances like the fact that allowing everyone to comment on every single core article is not the best strategy for all sites (typically you'd only want to allow that in specific categories; which categories depends on how you built your site). Therefore the idea of setting permissions automatically sounds good but the reality is that it's absolutely not a good idea at all to implement.

That's why I document things and tell you to read the documentation.

**Comment by breebee** [comment]
Ok the issue was due to FOF, so I uninstalled it, renamed plugins/system to plugins/system2 while fof error was blocking admin access, and reinstalled FOF4 directly from https://www.akeeba.com/download/fof4/4-0-4.html

I had forgotten to mention above that I had already had comments working prior to posting this issue rather then a lack of me reading documentation. Thank you Nikosdion for your patience and great contributions to the public community :)

**Comment by nikosdion** [comment]
First a general comment, if you please. I ask for accurate reproduction instructions so I can be sure it's a plausible bug. I am managing two dozen public projects and another dozen private projects. I have to prioritise. If I took a deep look at every reported issue I'd need 72 hours in a day just to handle those reports. If the reproduction instructions are inaccurate I get the wrong idea about what you have actually done and I might pass on your issue as most likely invalid. Please help me help you. Filling in the repro instructions takes you 5', saves me 2 hours and helps me address any real issues so much faster.

Regarding the issue, I was able to reproduce when another user sent me (via private email) the actual debugging information that appears with the 403 error when they set Debug Site to Yes. Even though they weren't sure they followed the documentation instructions correctly the debug information told me that something weird is going on with the FOF 4 Dispatcher. I addressed that issue and released a new version of FOF 4 _after_ I had already released Akeeba Engage 2.0.1.

Since Joomla picks up the FOF 4 updates there is no reason for me to publish another version of Akeeba Engage.

So, to recap: install Akeeba Engage 2.0.1, look for updates, an update to FOF 4 will be found, install it and your issue is resolved. I hope that makes sense.
